qa/tasks/cephfs/test_volumes.py

```diff
@@ -1941,6 +1941,53 @@ def test_subvolume_snapshot_ls(self):
         # verify trash dir is clean
         self._wait_for_trash_empty()
 
+    def test_subvolume_inherited_snapshot_ls(self):
```

Comment:
Need another test that snapshot info on an inherited snap fails.

Test you can't use `subvoluemm snapshot rm` on an inherited snap.

@ShyamsundarR does it make sense in CSI to have a subvolumegroup snapshot and subvolume snapshot with the same name? We can disambiguate them but it'd be better to not allow it to avoid confusion.

Reply:
> @ShyamsundarR does it make sense in CSI to have a subvolumegroup snapshot and subvolume snapshot with the same name? We can disambiguate them but it'd be better to not allow it to avoid confusion.

The names for snapshots (group or per-subvolume) in CSI land is a UUID, so conflicts should not arise. The UUID is generated by the CSI code, so it is feasible to ensure that we do not reuse one that is is the superset of subvolume and group snapshots already present.

There is no requirement or need to use the same names, even if the user decides to name a group snapshot and individual subvolume snapshots the same. As there is a (actually two) layer of mapping between the user provided names and UUIDs that we generate.

IOW, CSI should be good even if there is a hard restriction on naming conflict.

Reply:
@kotreshhr  is this resolved?

Reply:
@batrick You mean not allowing subvolumegroup and subvolume to be off same name ? I have not done anything w.r.t that.
I have added the tests for info and rm.

Reply:
> @batrick You mean not allowing subvolumegroup and subvolume to be off same name ? I have not done anything w.r.t that.

Yes. Let's prevent that in this patch.

Reply:
We have removed subvolumegroup snapshot creation from mgr/volumes for now. So this PR avoids subvolume snapshot creation with same name as it's subvolumegroup snapshot name if exists (subvolumegroup snapshot created by other means). 

Since subvolumegroup snapshot creation is not supported, the converse check doesn't hold (i.e., creation of subvolumegroup snapshot with same name as subvolume snapshot name).